C-Plus-Plus/sort/Bead_Sort.cpp

```diff
@@ -0,0 +1,84 @@
+/*Bead sort, also called as gravity sort, is a natural sorting algorithm since this algorithm was inspired from natural phenomenons
+and was designed keeping in mind objects(or beads) falling under the influence of gravity.The bead sort operation can be compared
+to the manner in which beads slide on parallel poles, such as on an abacus. However, each pole may have a distinct number of beads.*/
+
+#include <bits/stdc++.h>            // Header file to include STL library
+using namespace std;               //For cin and cout
+
+int main()
+{
+    vector<int> nums;                           // initialize a vector of integers
+    int a,n;                                    // Declaration of required variables
+    cout<<"Enter the number of elements :";
+    cin>>n;
+    cout<<"Enter the elements to be sorted : ";
+    for(int i=0;i<n;i++){                       //For loop for taking elements from the user, can be space-seperated as well
+        cin>>a;
+        nums.push_back(a);
+    }
+
+    nums.reserve(n);                        // reserve static vector for optimal performance gain
+
+    cout << "\nBefore sorting" << endl;    // Print vector before sorting
+    for(auto const& i: nums)
+        cout<<i<<" ";
+    cout<<"\n";
+
+    auto max = *max_element(nums.begin(), nums.end());      // find maximum element in vector
+
+    vector<unsigned char> beads(max * nums.size(), 0);       // declare beads vector for sorting
+
+    for(auto i = 0; i < nums.size(); i++)                   // initialize the beads accordingly
+        for(auto j = 0; j < nums[i]; j++)
```

Comment:
Push all your core code logic into another function. Keep the main() function solely for the driver code.